constraintlayout/core/src/main/java/androidx/constraintlayout/core/widgets/ConstraintWidget.java

```diff
@@ -2417,7 +2436,7 @@ public void addToSolver(LinearSystem system, boolean optimize) {
             }
         }
 
-        if (mVisibility == GONE && !hasDependencies()
+        if (mVisibility == GONE && !hasDependencies()  && !mAnimated
                 && !mIsInBarrier[HORIZONTAL] && !mIsInBarrier[VERTICAL]) {
```

Comment:
nit: compare with OR instead, where hasDependencies is the last check

Reply:
yeah good point -- can keep && if you prefer, but put it before !hasDependencies() :)

Reply:
For clarity I want this change to be as small as possible. Refactoring the line will obscure the logic change.

Reply:
I do not understand why before hasDependencies?
My original thinking was maintaining CL performance boost was the priority. i.e. 
Reducing this added check for only when the visibility check is gone and hasDependencies a small set. 
putting it after gone adds one more test to all gone widgets.

Reply:
How do you mean? I can't imagine the check having any impact in CL, which can't be said for MotionLayout since it has the opportunity to fail fast:
```
 if (mVisibillity == gone && !mAnimated && !hasDepedencies ...)
 
// !mAnimated is always false for MotionLayout
 
 if (mVisibillity == gone && false && !hasDepedencies ...)
 ```